crates/sui-adapter/src/execution_engine.rs

```diff
@@ -653,14 +654,21 @@ fn advance_epoch<S: ObjectStore + BackingPackageStore + ParentSync + ChildObject
             new_package.compute_object_reference()
         );
 
+        let write_kind = if version == OBJECT_START_VERSION {
```

Comment:
Can you also leave a comment in the place where we create a system package that says it's assumed here that the version must starts with OBJECT_START_VERSION?